en/03_Drawing_a_triangle/02_Graphics_pipeline_basics/01_Shader_modules.adoc

```diff
@@ -2,248 +2,362 @@
+Vulkan SDK. Make the script executable with `chmod +x compile.sh` and run it.
 
 *End of platform-specific instructions*
 
-These two commands tell the compiler to read the GLSL source file and output a SPIR-V bytecode file using the `-o` (output) flag.
-
-If your shader contains a syntax error then the compiler will tell you the line number and problem, as you would expect.
-Try leaving out a semicolon for example and run the compile script again.
-Also try running the compiler without any arguments to see what kinds of flags it supports.
-It can, for example, also output the bytecode into a human-readable format so you can see exactly what your shader is doing and any optimizations that have been applied at this stage.
-
-Compiling shaders on the commandline is one of the most straightforward options and it's the one that we'll use in this tutorial, but it's also possible to compile shaders directly from your own code.
-The Vulkan SDK includes https://github.com/google/shaderc[libshaderc], which is a library to compile GLSL code to SPIR-V from within your program.
+These two commands tell the compiler to read the SLang source file and output a
+SPIR-V 1.4 bytecode file directly using the `-o` (output) flag.
+
+Note: At the time of writing SlangC will natively support SPIR-V 1.3 and above
+without needing to go through emitting GLSL to get to SPIR-V.  While
+everything in this tutorial could work in SPIR-V 1.0, it would require us to
+break the SLang shaders up into multiple files which begs the question,
+what's the point? Plus, SPIR-V 1.4 starting from 1.4 means you'll be
+familiar with the latest the standard has to offer rather than starting from
+ an older version.
+
+If your shader contains a syntax error, then the compiler will tell you the line
+number and problem, as you would expect. Try leaving out a semicolon, for example,
+and run the compiler script again. Also try running the compiler without any
+arguments to see what kinds of flags it supports. It can, for example, also
+output the bytecode into a human-readable format, so you can see exactly what
+your shader is doing and any optimizations that have been applied at this stage.
+
+Compiling shaders on the commandline is one of the most straightforward
+options, yet the best path and one we use in this tutorial is to create a
+CMake function:
+
+[,cmake]
+----
+function (add_slang_shader_target TARGET)
+  cmake_parse_arguments ("SHADER" "" "SOURCES" ${ARGN})
+  set (SHADERS_DIR ${CMAKE_CURRENT_LIST_DIR}/shaders)
+  set (ENTRY_POINTS -entry vertMain -entry fragMain)
+  add_custom_command (
+          OUTPUT ${SHADERS_DIR}
+          COMMAND ${CMAKE_COMMAND} -E make_directory ${SHADERS_DIR}
+  )
+  add_custom_command (
+          OUTPUT  ${SHADERS_DIR}/slang.spv
+          COMMAND ${SLANGC_EXECUTABLE} ${SHADER_SOURCES} -target spirv -profile spirv_1_4 -emit-spirv-directly -fvk-use-entrypoint-name ${ENTRY_POINTS} -o slang.spv
```

Comment:
This seems to lack information on how SLANGC_EXECUTABLE is actually set/discovered.